File: GraphAdjacencyList.js
class Graph{

    //graph constructor initialiaing empty adjacecy list
    constructor() {
        this.adjacencyList = {};
    }
    
    //add vertex function
    insertVertex(node) {
        if(!this.adjacencyList[node]){ this.adjacencyList[node] = []; }
    }

    //add edge function
    insertEdge(source_node, destination_node){

        //if source node doesn't exist add it
        if(!this.adjacencyList[source_node]){
            this.insertVertex(source_node);
        }

        //if destination node doesn't exist add it
        if(!this.adjacencyList[destination_node]){
            this.insertVertex(destination_node);
        }

        //creating the connection s---->d
        this.adjacencyList[source_node].push(destination_node);

        //creating the connection s<----d
        this.adjacencyList[destination_node].push(source_node);
    }

    

    printAdjacencyList(){
        console.log(JSON.stringify(this.adjacencyList));
    }

    


}

File: GraphAdjacencyList.py
class Graph():
    adjacencyList = {}
    def __init__(self):
        self.adjacencyList = {}

    #add node
    def addNode(self, node):
        #check if node exists, if not then add that node
        if node not in self.adjacencyList:
            self.adjacencyList[node] = []

    #add edge
    def addEdge(self, snode, dnode):
        #check if source node exists, if not then add that node
        if snode not in self.adjacencyList:
            self.addNode(snode)
        #check if destination node exists, if not then add that node
        if dnode not in self.adjacencyList:
            self.addNode(dnode)

        # s---------->d
        self.adjacencyList[snode].append(dnode)

        # d<----------s
        self.adjacencyList[dnode].append[snode]
    
    # print adjacency list
    def printList(self):
        for key in self.adjacencyList.keys():
            print(key+" : "+self.adjacencyList[key])